cookbook/how_to_upgrade_to_new_version.rst

```diff
@@ -0,0 +1,224 @@
+
+1. Checkout from the GitHub Repository
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To retrieve a new version and upgrade your OroCRM instance, please execute the following steps:
+
+**1** Create backups of your Database and Code.
+
+**2** "Cd" to the crm root folder and switch the application to the maintenance mode.
+
+.. code-block:: bash
+
+    $ cd /path/to/application
+    $ sudo -uwww-data app/console lexik:maintenance:lock --env prod
+
+**3** Stop the cron tasks
+
+.. code-block:: bash
+
+    $ crontab -e -uwww-data
+
+    .. note::
+        www-data can be changed to user under which your web server runs
+
+Comment this line.
+
+.. code-block:: text
+
+     */1 * * * * /usr/bin/php /path/to/application/app/console --env=prod oro:cron >> /dev/null
+
+Kill the related job daemon process.
+
+.. code-block:: bash
+
+    $ ps ax|grep php5
+    $ kill <process_pid>
+
+<process_pid> - is a PID of currently executing application job daemon process. For example:
+
+.. code-block:: text
+
+    /path/to/application/app/console jms-job-queue:run --max-runtime=3600 --max-concurrent-jobs=5 --env=prod
+
+**4** Pull changes from the repository.
+
+.. code-block:: bash
+
+    $ sudo -uwww-data git pull
+    $ sudo -uwww-data git checkout <VERSION TO UPGRADE>
+
+**5** Upgrade the composer dependency and set correct owner to the retrieved files.
+
+.. code-block:: bash
+
+    $ sudo php composer.phar install --prefer-dist --no-dev
+    $ sudo chown www-data:www-data -R ./*
+
+**6** Remove old caches.
+
+.. code-block:: bash
+
+    $ sudo -u www-data app/console cache:clear --env prod
+
+or, as alternative
+
+.. code-block:: bash
+
+    $ sudo rm -rf app/cache/*
+
+**7** Upgrade the platform.
+
+.. code-block:: bash
+
+    $ sudo -u www-data php app/console oro:platform:update --env prod --force
+
+**8** Remove the caches.
+
+.. code-block:: bash
+
+    $ sudo -u www-data app/console cache:clear --env prod
+
+or, as alternative
+
+.. code-block:: bash
+
+    $ sudo rm -rf app/cache/*
+
+**9** Warm up the cahes
+
+.. code-block:: bash
+
+    $ sudo -u www-data app/console cache:warmup --env prod
+
+**10** Enable cron.
+
+.. code-block:: bash
+
+    $ crontab -e -uwww-data
+
+Uncomment this line.
+
+.. code-block:: text
+
+     */1 * * * * /usr/bin/php /path/to/application/app/console --env=prod oro:cron >> /dev/null
+
+**11** Switch back your application to normal mode from the maintenance mode.
+
+.. code-block:: bash
+
+    $ sudo -uwww-data app/console lexik:maintenance:unlock --env prod
+
+
+2. Download the Source Code Archive
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To retrieve a new version and upgrade your OroCRM instance, please execute the following steps:
+
+**1** Create backups of your Database and Code.
+
+**2** "Cd" to the crm root folder and switch the application to the maintenance mode.
+
+.. code-block:: bash
+
+    $ cd /path/to/application
+    $ sudo -uwww-data app/console lexik:maintenance:lock --env prod
+
+**3** Stop the cron tasks
```

Comment:
Should be checked is it necessary to stop the cron in case when we put system to the maintenance mode with app/console lexik:maintenance:lock. If I remember correctly cron will not perform any actions while maintenance mode is turned on.

According to this code https://github.com/orocrm/platform/blob/master/src/Oro/Bundle/CronBundle/Command/CronCommand.php#L45-L51 cron shouldn't run tasks in maintenance.